src/types/BulkUploadTask.ts

```diff
@@ -1,51 +1,44 @@
 import { ajv } from '../ajv';
 import { shortCodeSchema } from './ShortCode';
+import { idSchema } from './Id';
 import type { TaskStatus } from './TaskStatus';
 import type { JSONSchemaType } from 'ajv';
 import type { Writable } from './Writable';
 import type { Source } from './Source';
 import type { KeycloakId } from './KeycloakId';
 import type { ShortCode } from './ShortCode';
 import type { Funder } from './Funder';
+import type { File } from './File';
+import type { Id } from './Id';
 
 interface BulkUploadTask {
 	readonly id: number;
-	sourceId: number;
+	sourceId: Id;
 	readonly source: Source;
-	fileName: string;
-	sourceKey: string;
+	proposalDataFileId: Id;
```

Comment:
I think `fileId` would be sufficient here. The qualifier `proposalData` does not differentiate it from a file contained with a proposal. It could be qualified by `bulkUploadFileId` but then again we're inside a `BulkUploadTask` so that's redundant. :shrug:

Reply:
This name is necessary; we're about to add an additional file ID (`attachmentsArchiveFileID`)

Reply:
Added plural (this contains data for multiple proposals).  Data does stink, and maybe should be like `proposalsTable` (downside is sounds like SQL related) or `proposalsCsv`(downside is we may support tsv / xsl).

For now keeping it as data but renaming is incredibly easy so please don't hold back on continuing to offer iterative feedback.  This is worth getting right.